bake/bake.go

```diff
@@ -179,46 +180,82 @@ func dedupMap(ms ...map[string]string) map[string]string {
 	return res
 }
 
+func sliceToMap(env []string) (res map[string]string) {
+	res = make(map[string]string)
+	for _, s := range env {
+		kv := strings.SplitN(s, "=", 2)
+		key := kv[0]
+		switch {
+		case len(kv) == 1:
+			res[key] = ""
+		default:
+			res[key] = kv[1]
+		}
+	}
+	return
+}
+
 func ParseFiles(files []File, defaults map[string]string) (_ *Config, err error) {
 	defer func() {
 		err = formatHCLError(err, files)
 	}()
 
 	var c Config
-	var fs []*hcl.File
+	var composeFiles []File
+	var hclFiles []*hcl.File
+	envs := sliceToMap(os.Environ())
 	for _, f := range files {
-		cfg, isCompose, composeErr := ParseComposeFile(f.Data, f.Name)
-		if isCompose {
-			if composeErr != nil {
-				return nil, composeErr
+		switch filepath.Ext(strings.ToLower(f.Name)) {
+		case ".yml", ".yaml":
+			composeFiles = append(composeFiles, f)
```

Comment:
Like hcl and json, compose files need to be parsed later because we need to collect envs from files first.